test/service.js

```diff
@@ -0,0 +1,117 @@
+var sinon      = require('sinon'),
+request = require('request-promise'),
+Promise = require('bluebird'),
+Browserstack = require('../lib/browserstack-service.js');
+
+var getStub;
+var putStub;
+
+describe('BrowserstackService', function() {
+	const browserstack = new Browserstack;
+	before(function(){
+
+		global.browser = {
+			desiredCapabilities: {
+				'device': '',
+				'os': 'OS X',
+				'os_version': 'Sierra',
+				'browserName': 'chrome'
+			}
+		}
+
+		browserstack.auth = {};
+		browserstack.failures = 0;
+		browserstack.sessionId = 1;
+
+		putStub = sinon.stub(request,'put');
+		getStub = sinon.stub(request,'get');
+
+		putStub.resolves(Promise.resolve('{}'));
+		getStub.resolves(Promise.resolve('{}'));
+	});
+
+	after(function(){
+		request.put.restore;
```

Comment:
**IMPORTANT**: I think you'll need to call the restore function right? `request.put.restore();` There are a couple of these in this file.